src/test/java/teammates/test/cases/testdriver/TestNgTest.java

```diff
@@ -0,0 +1,114 @@
+package teammates.test.cases.testdriver;
+
+import static org.testng.AssertJUnit.assertTrue;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map.Entry;
+
+import org.testng.annotations.Test;
+
+import teammates.common.util.FileHelper;
+import teammates.test.cases.BaseTestCase;
+
+public class TestNgTest extends BaseTestCase {
+
+    @Test
+    public void checkTestsInTestNg() throws FileNotFoundException {        
+        String testNgXml = FileHelper.readFile("./src/test/testng.xml");
+        HashMap<String, String> testFiles = getTestFiles(testNgXml, "./src/test/java/teammates/test/cases"); // <class name, package name>
+        
+        List<String> filesExcludedFromTestNg = new ArrayList<String>();
+             
+        filesExcludedFromTestNg.add("BaseUiTestCase");         // Base*TestCase are base classes to be extended by the actual tests  
+        filesExcludedFromTestNg.add("FeedbackQuestionUiTest"); // Base class for all Feedback*QuestionUiTest (different question types)       
+        filesExcludedFromTestNg.add("GodModeTest");            // Needs to be run only when changes are made to GodMode
+        
+        testFiles = excludeFilesNotInTestNg(testFiles, filesExcludedFromTestNg);
```

Comment:
Things got a bit verbose here. May be using varargs is the way to go (see https://docs.oracle.com/javase/1.5.0/docs/guide/language/varargs.html).
```
testFiles = excludeFilesNotInTestNg(testFiles, 
                             "BaseUiTestCase",         // Base*TestCase are base classes to be extended by the actual tests  
                             "FeedbackQuestionUiTest", // Base class for all Feedback*QuestionUiTest (different question types)
                             "GodModeTest",            // Needs to be run only when changes are made to GodMode
                            );
  ```

The method needs to be like this:
`private HashMap<String, String> excludeFilesNotInTestNg(HashMap<String, String> testFiles, String... filesExcludedFromTestNg) { }`